inspirehep/modules/records/utils.py

```
@@ -44,3 +51,19 @@ def get_detailed_template_from_record(record):
     """Return the detailed template corresponding to the given record."""
     endpoint = get_endpoint_from_record(record)
     return current_app.config['RECORDS_UI_ENDPOINTS'][endpoint]['template']
+
+
+@contextmanager
+def file_resolving(fpath):
+    known_schemes = ['http', 'https']
+    if urlparse(fpath).scheme in known_schemes:
+        with closing(requests.get(url=fpath, stream=True)) as req:
+            if req.status_code == 200:
+                req.raw.decode_content = True
+                yield req.raw
+    else:
+        try:
+            with open(fpath, mode='r') as fh:
+                yield fh
+        except (OSError, IOError) as e:
+            LOGGER.error(e)
```

it should probably not swallow the exception here